Dockerfile

```diff
@@ -40,7 +55,8 @@ ENV OMPI_ALLOW_RUN_AS_ROOT=1
 ENV OMPI_ALLOW_RUN_AS_ROOT_CONFIRM=1
 
 # set number of OpenMP threads to 1 by default
-ENV OMP_NUM_THREADS=1
+# Only during test phase because users will most likely customize it when running actual jobs. -Jason
+ARG OMP_NUM_THREADS=1
```

Comment:
The problem is that if a new user (who doesn't know about this ) doesn't set the `OMP_NUM_THREADS` variable then Octopus will most definitely over subscribe leading to poor performance.
By setting `OMP_NUM_THREADS` by default as 1. an experienced user can always overwrite the value but the default user runs would not have a performance hit.  This would simplify the user workflow for eg in binder or other situations where we just introduce octopus.

Reply:
I see, thanks for your ideas!
I could see a docker file with cuda configured being useful. 
Could we maybe keep two separate docker files, one for without GPU support ( the default now) which is used for octopus training and as a reference the present one based on ubuntu which could serve as good reference.

Reply:
I think it is a good idea to have it. For future reference at least.

I just worked on a different recipe that based on Intel OneAPI (instead of GCC) and manually installed CUDA driver & toolkit. Because a user reported to us that, at least on our Intel-based clusters, Intel-compiled Octopus are ~20% faster than GCC-compiled one (for reasons I do not fully understand yet). So I made an Intel-based recipe instead. I will create another branch as well for your references. The official Intel OneAPI container is also Ubuntu-based.